### ISSUE TITLE
No devices found under this Xiaomi account

### ISSUE THREAD

**Issue by auanasgheps** [al-one/hass-xiaomi-miot#1956]
### Device model / 设备型号

dreame.vacuum.p2029

### Component version / 插件版本

1.0.0b3

### HA core version / HA版本

2024.11.2

### Integrated mode / 集成方式

Local (本地模式)

### The problem / 问题详情

Hi,
I've been using this integration for a long time but now I'm facing an issue: my vacuum is not available.
However, I think there's an issue with the login/account part because that's what I get If I try to reconfigure the integration.

![immagine](https://github.com/user-attachments/assets/cbd7d986-69c7-4c5f-86f9-ad8335cfe490)
![immagine](https://github.com/user-attachments/assets/6f0ff912-b07e-4f47-a5dd-a4cb2abab7c3)

I usually set it to "Local", I also tried Automatic and Cloud but no change.

I have 3 devices on my accout on such server, just verified by re-authenticating the app on my phone.

The integration has stopped working since the 16th november.

I was using the latest stable. I just updated to the beta, no difference.

I tried loggin in with email and user ID, no difference.

### Entity attributes / 实体属性

```yaml
"unavailable"
```

### Home Assistant Logs / 系统日志

2024-11-18 12:23:48.947 WARNING (MainThread) [custom_components.xiaomi_miot.core.xiaomi_cloud] Got xiaomi devices for <REDACTED> failed: {'code': 0, 'message': 'SERVICETOKEN_EXPIRED'}
2024-11-18 12:23:51.232 ERROR (SyncWorker_7) [miio.miioprotocol] Got error when receiving: {'code': -9999, 'message': 'user ack timeout'}

**Comment by al-one** [comment]
Try master branch.

**Comment by rasasak** [comment]
same issue on 0.7.22 version 

xiaomi connected devices stop working ...  when I try reset with force .option  = no device found ... 

on xiaomi home app .. is device conected and working ...  but in this integration suddenly stop  responsing

**Comment by al-one** [comment]
```bash
wget -O - https://get.hacs.vip | DOMAIN=xiaomi_miot ARCHIVE_TAG=master bash -
```

**Comment by auanasgheps** [comment]
Thanks. I've ran this command and rebooted HA, but the version installed seems to be: 0.7.23


![immagine](https://github.com/user-attachments/assets/bf79947f-9ad1-470e-a44a-b101f1a15623)


By the way, no difference, the setup errors out and I still get the exact same error.

**Comment by TheSnowfield** [comment]
> * [登录失败/没有实体等常见问题解决办法 #500 (comment)](https://github.com/al-one/hass-xiaomi-miot/issues/500#issuecomment-2482015695)

It complains  `{"code":0,"message":"invalid signature"}`
```
2024-11-18 22:27:54.745 WARNING (MainThread) [custom_components.xiaomi_miot.core.xiaomi_cloud] Error while executing request to https://api.io.mi.com/app/v2/message/v2/typelist: {"code":0,"message":"invalid signature"}
2024-11-18 22:27:54.748 WARNING (MainThread) [custom_components.xiaomi_miot.sensor] Get xiaomi message for xx failed: {'code': 0, 'message': 'invalid signature'}
2024-11-18 22:27:55.030 WARNING (MainThread) [custom_components.xiaomi_miot.core.xiaomi_cloud] Error while executing request to https://api.io.mi.com/app/v2/homeroom/gethome_merged: {"code":0,"message":"invalid signature"}
2024-11-18 22:27:55.102 WARNING (MainThread) [custom_components.xiaomi_miot.core.xiaomi_cloud] Error while executing request to https://api.io.mi.com/app/home/device_list: {"code":0,"message":"invalid signature"}
2024-11-18 22:27:55.103 WARNING (MainThread) [custom_components.xiaomi_miot.core.xiaomi_cloud] Got xiaomi devices for my@example.com failed: {'code': 0, 'message': 'invalid signature'}
2024-11-18 22:27:55.177 WARNING (MainThread) [custom_components.xiaomi_miot.core.xiaomi_cloud] Error while executing request to https://api.io.mi.com/app/scene/history: {"code":0,"message":"invalid signature"}
2024-11-18 22:27:55.178 WARNING (MainThread) [custom_components.xiaomi_miot.sensor] Get xiaomi scene history for xx xx failed: {'code': 0, 'message': 'invalid signature'}
2024-11-18 22:27:55.196 WARNING (MainThread) [custom_components.xiaomi_miot.core.xiaomi_cloud] Error while executing request to https://api.io.mi.com/app/scene/history: {"code":0,"message":"invalid signature"}
2024-11-18 22:27:55.197 WARNING (MainThread) [custom_components.xiaomi_miot.sensor] Get xiaomi scene history for xx xx failed: {'code': 0, 'message': 'invalid signature'}
2024-11-18 22:27:55.422 WARNING (MainThread) [custom_components.xiaomi_miot.core.xiaomi_cloud] Error while executing request to https://api.io.mi.com/app/scene/history: {"code":0,"message":"invalid signature"}
2024-11-18 22:27:55.424 WARNING (MainThread) [custom_components.xiaomi_miot.sensor] Get xiaomi scene history for xx xx failed: {'code': 0, 'message': 'invalid signature'}
```

**Comment by al-one** [comment]
Relogin

**Comment by TheSnowfield** [comment]
> Relogin

I selected `Auto Mode` and `更新设备列表` and submitted the login request, then got this log.

**Comment by al-one** [comment]
Try master branch.
```bash
wget -O - https://get.hacs.vip | DOMAIN=xiaomi_miot ARCHIVE_TAG=master bash -
```

**Comment by nelsongraca** [comment]
same issue here, tried master branch but it's asking for a captcha, REMOVED is because I removed the actual code

    ⚠️ Login to xiaomi error: &&&START&&&{"code":87001,"desc":"验证码输入错误","captchaUrl":"/pass/getCode?icodeType=login&0.REMOVED","type":"manMachine"}

**Comment by domonkos11** [comment]
Same problem. api.io.mi.com says 404 when doing curl, could that be a problem?

**Comment by Pixelw** [comment]
Same issue with updated version 0.7.23
Got xiaomi devices for XXXX failed: {'code': 0, 'message': 'SERVICETOKEN_EXPIRED'}

**Comment by al-one** [comment]
master branch ...

**Comment by wespen** [comment]
Can you bump the version so it pulls update in HACS

**Comment by baskinsy** [comment]
Removing and re-adding the account worked for me.

**Comment by wespen** [comment]
I tried it...and it worked.
I deleted the old account and re-added it and now it works.

**Comment by javierbrea** [comment]
Automatically solved when installing [v0.7.24](https://github.com/al-one/hass-xiaomi-miot/releases/tag/v0.7.24). Thank you!
